src/lib.rs

```diff
@@ -180,11 +182,10 @@ impl Client {
             .ok_or_else(|| RedirectedError::AuthFailed(String::new()))?;
 
         let token = self
-            .token_request(&[
-                ("grant_type", "authorization_code"),
-                ("code", code),
-                ("redirect_uri", &url[..url::Position::AfterPath]),
-            ])
+            .token_request(TokenRequest::AuthorizationCode {
+                code: code.borrow(),
```

Comment:
`Deref` is probably more appropriate here, as it's the method mentioned in the `Cow` docs and it avoids confusion with other methods named `borrow`.